ste/mgr-JobMgr.go

```diff
@@ -313,6 +399,21 @@ func (jm *jobMgr) AddJobPart(partNum PartNumber, planFile JobPartPlanFileName, s
 	jm.finalPartOrdered = jpm.planMMF.Plan().IsFinalPart
 	jm.setDirection(jpm.Plan().FromTo)
 	jpm.exclusiveDestinationMap = jm.getExclusiveDestinationMap(partNum, jpm.Plan().FromTo)
+
+	// This is actually a really fantastic place to set up our user delegation authentication manager.
+	// This is because we have no awareness of our source root until now,
+	// and two job parts will never have different sourceroots.
+	// Note that setupUDAM checks if UDAM is nil before attempting to create it.
+	if jpm.Plan().FromTo.IsS2S() && jpm.Plan().FromTo.From() == common.ELocation.Blob() && len(jpm.sourceSAS) == 0 {
+		// So now you may be thinking "Can't you copy safely from a public resource?"
+		// The answer is yes.
+		// If a user is copying from a public resource in this case, one of two things will happen:
+		// 1) setupUDAM will fail cleanly, leaving an empty UDAM instance that does nothing.
```

Comment:
TODO: Consider the implications of a dummy UDAM instance

Reply:
Dummy UDAM instances are _ok_, but we should add some extra logging and stdout warnings for it.